ampd/src/tofnd/grpc.rs

```diff
@@ -68,33 +66,32 @@ impl Multisig for MultisigClient {
             .clone()
             .keygen(request)
             .await
-            .and_then(|response| {
-                response
-                    .into_inner()
+            .map_err(ErrorExt::into_report)
+            .and_then(|res| {
+                let res = res.into_inner();
+
+                res.clone()
                     .keygen_response
-                    .ok_or_else(|| Status::internal("keygen response is empty"))
+                    .ok_or(report!(Error::InvalidKeygenResponse).attach(res))
```

Comment:
see comment about debug hooks. You can also add additional values to `instrument`, see https://docs.rs/tracing/latest/tracing/attr.instrument.html#adding-fields